zappa/cli.py

```diff
@@ -1789,18 +1788,36 @@ def execute_prebuild_script(self):
 
         """
 
-        # Parse the string
-        prebuild_module_s, prebuild_function_s = self.prebuild_script.rsplit('.', 1)
+        (pb_mod_path, pb_func) = self.prebuild_script.rsplit('.', 1)
 
-        # The module
-        prebuild_module = imp.load_source(prebuild_module_s, prebuild_module_s + '.py')
+        try:  # Prefer prebuild script in working directory
+            if pb_mod_path.count('.') >= 1:  # Prebuild script func is nested in a folder
+                (mod_folder_path, mod_name) = pb_mod_path.rsplit('.', 1)
+                mod_folder_path_fragments = mod_folder_path.split('.')
+                working_dir = os.path.join(os.getcwd(), *mod_folder_path_fragments)
+            else:
+                mod_name = pb_mod_path
+                working_dir = os.getcwd()
+
+            working_dir_importer = pkgutil.get_importer(working_dir)
+            module_ = working_dir_importer.find_module(mod_name).load_module(mod_name)
 
-        # The function
-        prebuild_function = getattr(prebuild_module, prebuild_function_s)
+        except (ImportError, AttributeError):
+
+            try:  # Prebuild func might be in virtualenv
+                module_ = importlib.import_module(pb_mod_path)
+            except ImportError:  # pragma: no cover
+                raise ClickException(click.style("Failed ", fg="red") + 'to ' + click.style(
+                    "import prebuild script ", bold=True) + 'module: "{pb_mod_path}"'.format(
+                    pb_mod_path=click.style(pb_mod_path, bold=True)))
 
-        # Execute it
-        prebuild_function()
+        if not hasattr(module_, pb_func):  # pragma: no cover
+            raise ClickException(click.style("Failed ", fg="red") + 'to ' + click.style(
+                "find prebuild script ", bold=True) + 'function: "{pb_func}" '.format(
+                pb_func=click.style(pb_func, bold=True)) + 'in module "{pb_mod_path}"'.format(
+                pb_mod_path=pb_mod_path))
 
+        getattr(module_, pb_func)()  # Call the function
```

Comment:
Can we split this out, as in the original? Having the `getattr()` on a separate line from the invocation makes it easier when debugging from a stack trace to see if it was the getattr() that failed or the invocation that failed.